lib/syntax.ml

```diff
@@ -1,3 +1,20 @@
+open Compat
+
 type t = Normal | Cram
 
 let cram_default_header = Some "sh" (* FIXME: bash? *)
+
+let pp fs = function
+  | Normal -> Fmt.string fs "normal"
+  | Cram -> Fmt.string fs "cram"
+
+let equal x y = match (x, y) with
+  | Normal, Normal
+  | Cram, Cram -> true
+  | _ -> false
+
```

Comment:
Isn't this the standard behavior of OCaml's `=` operator ?

Reply:
Anyway, if this is going to stay, the `false` case should specify all the constructors: `| (Normal | Cram), _ -> false`.
This ensures a compilation error if a constructor is added.